docs/debug.md

```diff
@@ -0,0 +1,150 @@
+# Debugging SOCI snapshotter
+
+This document outlines where to find/access logs and metrics for the snapshotter. It attempts to provide some common error paths that a user might run into while using the snapshotter and provides some guidance on how to either root-cause the issue or resolve it.
+- [Debugging SOCI snapshotter](#debugging-soci-snapshotter)
+  - [Finding Logs / Metrics](#finding-logs--metrics)
+    - [Logs](#logs)
+    - [Metrics](#metrics)
+      - [Accessing Metrics](#accessing-metrics)
+      - [Metrics Emitted](#metrics-emitted)
+  - [Common Scenarios](#common-scenarios)
+    - [Rpull](#rpull)
+      - [No lazy-loading](#no-lazy-loading)
+      - [Pull taking an abnormal amount of time](#pull-taking-an-abnormal-amount-of-time)
+      - [Background Fetching](#background-fetching)
+    - [Running container](#running-container)
+      - [FUSE Read Failures](#fuse-read-failures)
+
+
+
+## Finding Logs / Metrics
+
+### Logs
+
+Before debugging it is important to first establish what logs would be most useful and where they can be located. For the most part, the SOCI grpc logs will be the most important place to look when debugging. If the SOCI snapshotter grpc server was started via `systemd` than you can obtain logs using `journalctl`. 
+
+`journalctl -u soci-snapshotter-grpc.unit`
+
+If you have started SOCI-snapshotter grpc manually, logs will either be emitted to stderr/stdout or to the destination of your choice.
+
+### Metrics
+
+#### Accessing Metrics
+
+All metrics are collected via Prometheus and can be accessed by first setting the `metrics_address` field within SOCI’s `config.toml` file found within `/etc/soci-snapshotter-grpc` . You can provide any local address as long as the port is not in use. To view the metrics you can send a `GET` request via any HTTP client to the `/metrics` endpoint and Prometheus will return a list of all the metrics emitted by the snapshotter.
+```
```

Comment:
one empty line before and after code block. 

also line 35 add `sheel` language annotation.

If a code block has both commands and outputs, usually I put a $ before the command to distinguish them.